generator/java/resources/templates/ApiClient.mustache

```diff
@@ -1508,15 +1508,24 @@ public class ApiClient {
     }
 
     private void refreshTokenIfNotValid() throws ApiException {
-        if (tokenInfo == null || !tokenInfo.isValid()) {
-            HttpBasicAuth auth = (HttpBasicAuth)authentications.get("CriteoHttpBasicAuth");
-            if (auth.getUsername() == null || auth.getPassword() == null) {
-                throw new IllegalArgumentException("username or password is not present.");
-            }
-            AccessTokenModel response = (new OAuthApi(this)).getToken(
-                "client_credentials", auth.getUsername(), auth.getPassword(), null, null, null);
-            tokenInfo = new TokenInfo(response.getExpiresIn() + System.currentTimeMillis() / 1000);
-            this.setApiKey(response.getAccessToken());
+        OAuth oauth = (OAuth) this.getAuthentication("oauth");
+        String accessToken = oauth.getAccessToken();
+
+        if (accessToken != null) {
+            return;
+        }
```

Comment:
Could the accessToken be invalid or expired? Should we trigger a refresh in this case?

Reply:
This case is when the accessToken is manually set by the user. We can't know if the token is invalid or expired, we can just trust the developer and use its provided token.